File: src/main/java/ru/citeck/idea/files/NavigateInFileItemsProvider.java
package ru.citeck.idea.files;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.psi.PsiFile;
import org.jetbrains.annotations.Nullable;

import java.util.Collection;

public interface NavigateInFileItemsProvider {

    ExtensionPointName<NavigateInFileItemsProvider> EP_NAME =
        ExtensionPointName.create("ru.citeck.idea.navigateInFileItemsProvider");

    @Nullable Collection<NavigateInFileItem> getItems(PsiFile psiFile);


}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/navigate/providers/NavigateInFileItemsProvider.kt
package ru.citeck.idea.artifacts.action.navigate.providers

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.psi.PsiFile

interface NavigateInFileItemsProvider {

    companion object {
        val EP_NAME: ExtensionPointName<NavigateInFileItemsProvider> =
            ExtensionPointName.create("ru.citeck.idea.navigateInFileItemsProvider")
    }

    fun getItems(psiFile: PsiFile?): List<NavigateInFileItem>?
}
